docs/cluster-scope/add_resource_to_cluster.md

```diff
@@ -0,0 +1,24 @@
+# Adding Cluster Resources
+
+To add a cluster resource to a cluster, you will need to know the `${ENVIRONMENT}` and `${CLUSTER}` you would like to add this resource to.
+
+You will also need the url to the raw file (e.g. a raw file from github) or a path to the resource manifest on your local file system.
+
+You will also need to be logged in to the cluster that supports this api resource. This is done to fetch the plural name of this resource from the cluster, no actual changes are made to the live environment.
```

Comment:
I am a little sad that this solution requires:

1. A pre-existing file
1. A network connection
1. Active cluster authentication

That's a lot of prerequisites. The first item, in particular, really upends the typical model (create the file in the appropriate location, then `git add` when done).

I guess my preference here would be to see the organization enforced by a precommit check, and allow the script to be run without requiring remote connectivity (so if I give it the wrong metadata and create a file in the wrong place, the precommit check will yell at me either locally or when I submit the pr).

Reply:
If we're only using the remote connection to determine the plural of a resource name, what if just cache that in the repository? The available api resources isn't going to change all that often, so we can update the cached version when necessary.

Reply:
> That's a lot of prerequisites. The first item, in particular, really upends the typical model (create the file in the appropriate location, then git add when done).

This wouldn't really be for that particular usecase, but for more situations where we get requests from users to add cluster-resources, like: https://github.com/operate-first/apps/pull/588

Here we are [asked by this particular team](https://github.com/operate-first/support/issues/192#issuecomment-825206787) to add a bunch of resources found in their git repo, having to create kustomization/new files for all these files are made easier if I can just feed the link and have it be auto created. 

The manual process of adding your own file would still be an option. I agree we can also add ci tests to ensure integrity, and we should. 

> If we're only using the remote connection to determine the plural of a resource name, what if just cache that in the repository? The available api resources isn't going to change all that often, so we can update the cached version when necessary. 

Yeah we can do that.